pkg/controller/metrics/metrics.go

```diff
@@ -124,16 +191,17 @@ func init() {
 	metrics.Registry.MustRegister(metricSyncSetsUnappliedTotal)
 	metrics.Registry.MustRegister(MetricControllerReconcileTime)
 
-	metrics.Registry.MustRegister(MetricClusterDeploymentProvisionUnderwaySeconds)
 	metrics.Registry.MustRegister(MetricClusterDeploymentDeprovisioningUnderwaySeconds)
 }
 
 // Add creates a new metrics Calculator and adds it to the Manager.
 func Add(mgr manager.Manager) error {
 	mc := &Calculator{
-		Client:   mgr.GetClient(),
-		Interval: 2 * time.Minute,
+		Client:        mgr.GetClient(),
+		Interval:      2 * time.Minute,
+		CustomMetrics: newCustomCollector(mgr.GetClient()),
```

Comment:
Lets rename this var as well, and it can be private and not exported it looks like.

It's a little strange we carry this on the metrics controller when it isn't really used there, I would have expected an init block and a global package variable, but that is harder to get a client to.

Reply:
I considered adding the collector to the manager as Matthew suggested, however manager can only add [Runnable](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/manager#Runnable), which would require prometheus [Collector](https://godoc.org/github.com/prometheus/client_golang/prometheus#Collector) to implement Runnable's Start method.
We have to instantiate and register the custom collector when the manager client is available, so Add method seems to be the appropriate place. I did remove it from the Calculator struct though, since we don't use the object itself anywhere else.

Reply:
The `Add` method should get the client and save it in the object that is being added. Then the `Start` method should be registering the metric.

Reply:
But I am not strongly opposed to the approach that is being taken now. Registering the metrics in init blocks is already pervasive throughout the code, so adding to that anti-pattern is not a deal-breaker for me.

Reply:
> The `Add` method should get the client and save it in the object that is being added. Then the `Start` method should be registering the metric.

However, the custom collector is instantiating a prometheus.Collector... So the Add and Start methods need to have been implemented by prometheus.Collector.